pwndbg/gdblib/kernel/__init__.py

```diff
@@ -101,30 +120,43 @@ def phys_to_pfn(self, phys: int) -> int:
     def pfn_to_phys(self, pfn: int) -> int:
         raise NotImplementedError()
 
-    def phys_to_page(self, phys: int) -> int:
+    def pfn_to_page(self, phys: int) -> int:
         raise NotImplementedError()
 
-    def page_to_phys(self, page: int) -> int:
+    def page_to_pfn(self, page: int) -> int:
         raise NotImplementedError()
 
+    def virt_to_pfn(self, virt: int) -> int:
+        return phys_to_pfn(virt_to_phys(virt))
+
+    def pfn_to_virt(self, pfn: int) -> int:
+        return phys_to_virt(pfn_to_phys(pfn))
+
+    def phys_to_page(self, phys: int) -> int:
+        return pfn_to_page(phys_to_pfn(phys))
+
+    def page_to_phys(self, page: int) -> int:
+        return pfn_to_phys(page_to_pfn(page))
+
     def virt_to_page(self, virt: int) -> int:
-        return phys_to_page(virt_to_phys(virt))
+        return pfn_to_page(virt_to_pfn(virt))
 
     def page_to_virt(self, page: int) -> int:
-        return phys_to_virt(page_to_phys(page))
+        return pfn_to_virt(page_to_pfn(page))
 
 
 class x86_64Ops(ArchOps):
     def __init__(self) -> None:
-        if "X86_5LEVEL" in kconfig() and "no5lvl" not in kcmdline():
-            raise NotImplementedError("Level 5 page table support is not implemented")
+        if uses_5lvl_paging():
+            self.PAGE_OFFSET = 0xFF11000000000000
```

Comment:
Can we add a comment where do these values come from? Ideally pointing out to kernel sources where this is defined (I guess this is where it is defined?:P)

Reply:
I just added the references